src/main/java/gregtech/common/ClientProxy.java

```diff
@@ -182,6 +190,37 @@ public static void addMaterialFormulaHandler(ItemTooltipEvent event) {
             if (chemicalFormula != null && !chemicalFormula.isEmpty())
                 event.getToolTip().add(1, ChatFormatting.GRAY.toString() + chemicalFormula);
         }
+
+        // Quantum Tank/Chest NBT Clearing Recipe Tooltip
+        final EntityPlayer player = event.getEntityPlayer();
+        InventoryCrafting inv = null;
+        if (player != null) {
+
+            if (player.openContainer instanceof ContainerWorkbench)
+                inv = ((ContainerWorkbench) player.openContainer).craftMatrix;
+            else if (player.openContainer instanceof ContainerPlayer)
+                inv = ((ContainerPlayer) player.openContainer).craftMatrix;
+
+            if (inv != null) {
+                boolean foundSelf = false;
+                for (int i = 0; i < inv.getSizeInventory(); i++) {
+                    if (ItemStack.areItemsEqual(inv.getStackInSlot(i), event.getItemStack())) {
+                        foundSelf = true;
```

Comment:
btw, before in the method `ItemStack itemStack = event.getItemStack()` was declared so doing `event.getItemStack()` everytime is not necessary.
```suggestion
                        if (itemStack.getItem() instanceof MachineItemBlock) {
                            if (MachineItemBlock.getMetaTileEntity(itemStack).keepsInventory()) {
                                event.getToolTip().add(I18n.format("gregtech.universal.clear_nbt_recipe.tooltip"));
                            }
                        }
```

Also, IMO the tooltip should only be in the resulting item, you did this before with `craftResult.getStackInSlot(0) == itemStack`. But as I said personal opinion.

Reply:
This suggestions is not entirely correct as not every `MachineItemBlock` with `keepsInventory() ==true` has recipe to remove NBT data

Reply:
If they don't have the recipe, the checks should fail before, part of the reason I suggested the tooltip to only be in the resulting item.

Reply:
There is nothing which prevent to have recipe with same input as output which does not clear NBT so in case of `keepsInventory() == true` it would assume that it is NBT clearing recipe which is not correct. Sure that chance is not big but someone may have these recipes generated automatically by other mod...I would be rather more defensive regarding this.